### ISSUE TITLE
TypeScript Language Server and editor aren't synced correctly

### ISSUE THREAD

**Issue by donaldpipowitch** [Microsoft/vscode#57860]
Hi there,

my colleagues (cc @otbe) and I have some problems with the TypeScript Language Server. The problem appeared maybe around one or two months ago. We experienced the same issue occasionally independent from each other and thought it was _our_ setup, before we actually noticed that multiple developers here had the same issue. We thought it would be fixed in one of the next VS Code (Insiders) appears, because we thought it must was introduced by an earlier update. But after some weeks and updates the error is still here. So I began looking for problems like ours in the issue tracker, but besides several "TypeScript Language Server is slow" issues I haven't found something which matched our problem.

So without further ado, here is the problem:
We experience what looks like that VS Code and the TypeScript Language Server become out-of-sync. This doesn't happen every time, but once an hour or so. The errors than look like this and we need to restart the TS server to fix them.

![image](https://user-images.githubusercontent.com/1152805/45029782-0dc6c900-b04a-11e8-9f3a-87d9142a6967.png)

Sadly we can't provide more information as we don't know why this happens 😔

- VSCode Version: 1.27.0-insider (7b9afc4196bda60b0facdf62cfc02815509b23d6)
- OS Version: macOS 10.13.6

Steps to Reproduce:

1. Nothing deterministic. Sorry :(

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Sounds like https://github.com/Microsoft/TypeScript/issues/25630

Are you using any typescript server plugins in your workspace?

**Comment by otbe** [comment]
Yes and no.
The picture is taken from a project wihtout any typescript server plugins but in other projects we use the [typescript-styled-plugin](https://github.com/Microsoft/typescript-styled-plugin).

**Comment by mjbvz** [comment]
How are you using the plugins? Through the VS Code extension or installed with npm?

**Comment by donaldpipowitch** [comment]
Both I guess. It's added to our `tsconfig.json`, but some also use the VS Code extension for syntax highlighting 🤔

**Comment by mjbvz** [comment]
Can you please try using vscode with this plugin fully disabled and check if you still see the issue?

**Comment by 19majkel94** [comment]
I experience the same problem. I was thinking that's due to the big and complex project but then it started to happen also in my other projects:

![2018-09-07_13-14-29](https://user-images.githubusercontent.com/10618781/45216838-3ba64a80-b2a2-11e8-8a30-1547f1a592bb.gif)

I've checked and disabled the plugin and extension, even in VSCode insider - still the same lag 😕 

Can I enable some verbose log or profiler to dump some info for you to debug?

**Comment by donaldpipowitch** [comment]
Is your problem a (more or less) constant lag, but which resolves after some delay? Because our out-of-sync problems stay no matter how long we wait. It doesn't recover from itself, only be restarting the service or when we start typing more it eventually gets in-sync again.

**Comment by 19majkel94** [comment]
@donaldpipowitch Yes, it's bigger or smaller but it's a constant lag. Sometimes it hangs for too long so I have to do `CTRL + A`, `CTRL + X` and `CTRL + V` combination to force checking whole file again and then the red errors comes away.

I also have launched `npx tsc --noEmit --watch` parallel in terminal and it's working super fast, only VSCode presenting is laggy.

**Comment by quicksnap** [comment]
I am also having lag problems similarly described. TS Version 3.0.3, VSCode 1.27.1 (just updated to .2, can test again soon)

I noticed the problem is mitigated when I disable `vscode-styled-components` plugin, but I have a feeling that only alleviates a more general problem. Happy to provide more info if needed.

**Comment by quicksnap** [comment]
Same with my coworkers, too. We don't have a _huge_ codebase but we have a decent amount of files and types.

**Comment by kingdaro** [comment]
[Here's a recorded instance of this happening.](https://streamable.com/8kgb7) As you can see, the `storedChannelsKey` _is_ being used, but TS doesn't think it is. The `thisthis` errors reported aren't present in my code, but they were before I deleted a period or two.

Everything goes back to normal if I "wake it up" by typing and deleting a space, saving the file, formatting the document, etc.

**Comment by 19majkel94** [comment]
Looks like it's resolved on TS side:
https://github.com/Microsoft/typescript-template-language-service-decorator/commit/387cd5cee6a9c086dc6c887b81c6034781f8fc76

Can't wait for a patch release... using VSCode with TS became a nightmare recently in bigger projects.

**Comment by donaldpipowitch** [comment]
I updated the TS plugin today and everything looked good ♥️

**Comment by mjbvz** [comment]
For verification:

1. Install the styled and lit-html vs code extensions.
1. Then follow steps in https://github.com/Microsoft/TypeScript/issues/25630
